assembly/src/parsers/ast/context.rs

```diff
@@ -0,0 +1,282 @@
+use super::{
+    nodes::Procedure, parse_op_token, AssemblyError, Instruction, Node, ProcMap, Token,
+    TokenStream, MODULE_PATH_DELIM,
+};
+use vm_core::utils::{
+    collections::{BTreeMap, Vec},
+    string::{String, ToString},
+};
+
+// Context
+// ================================================================================================
+
+// AST Parser context that holds internal state to generate correct ASTs.
+pub struct ParserContext {
+    pub imports: BTreeMap<String, String>,
+    pub procedures: ProcMap,
+}
+
+impl ParserContext {
+    // Parses an if-else statement from the provided token stream.
```

Comment:
nit: I'd probably create a couple sections grouping related functions together to improve readability. For example `STATEMENT PARSERS` (for if/while/repeat statements), `PROCEDURE PARSERS`, `CALLL PARSERS` (for call/exec instructions).

Reply:
fixed